Gems/SceneProcessing/Code/Source/Generation/Components/UVsGenerator/UVsPreExportComponent.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <SceneAPI/SceneCore/Components/GenerationComponent.h>
+#include <SceneAPI/SceneCore/Containers/Scene.h>
+
+#include <AzCore/RTTI/RTTI.h>
+
+namespace AZ::SceneAPI::Events { class GenerateAdditionEventContext; }
+
+namespace AZ::SceneGenerationComponents
+{
+    //! This is the component responsible for actually hooking into the scene API's processing flow
+    //! during the generation step.
+    class UVsPreExportComponent
+        : public AZ::SceneAPI::SceneCore::GenerationComponent
+    {
+    public:
+        AZ_COMPONENT(UVsPreExportComponent, "{64F79C1E-CED6-42A9-8229-6607F788C731}", AZ::SceneAPI::SceneCore::GenerationComponent)
```

Comment:
FYI we have a new AZ_COMPONENT_DECL and AZ_COMPONENT_IMPL macros that allows the move the inline definitions of Component Descriptor, TypeInfo, RTTI and Allocator functions to the cpp file.

That could save a tiny bit of compile time.

Reply:
Looking at the code, everything in scene api works by trawling the serialize context for classes, there's no reason for any of these components to even show up in the headers at all.  I'll move the entire components to the impl files, that way they'll be in just 1 compile unit.